week-06/day-4/WebShop/WebShop/Controllers/ShopController.cs

```diff
@@ -56,5 +56,16 @@ public IActionResult MostExpensiveAvailable()
             ViewData["expensiveAvailable"] = MostExpensiveAvailableItem.Name;
             return View();
         }
+
+        [HttpPost]
+        public IActionResult SearchBar(string search = null)
+        {
+            if (!string.IsNullOrEmpty(search))
+            {
+                var searchName = ShopItemList.Where(s => s.Name.Contains(search) || s.Description.Contains(search)).ToList();
```

Comment:
`searchName` is not a good variable name for this. We get a subset of `shopItems`. So `filteredShopItems` would be a nice name.